lib/repositories/genre/Impl/genre_from_json.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:convert';
+import 'package:flutter/services.dart';
+
+import '../i_genre_repository.dart';
+
+class GenresFromJson extends GenresRepository {
+  late final Map<int, String> genresById;
+  static const String genresPath = 'jsonData/genres.json';
+
+  GenresFromJson() {
+    genresById = {};
+  }
+
+  @override
+  Future<String?> getGenre(int id) async {
+    if (genresById.isEmpty) {
+      genresById.addAll(await parseJsonGenres());
+    }
+    return genresById[id];
+  }
+
+  static Future<Map<int, String>> parseJsonGenres() async {
+    var jsonPath = await rootBundle.loadString('jsonData/genres.json');
+    Map<String, dynamic> json = jsonDecode(jsonPath);
+    return {for (var e in json['genres']) e['id'] as int: e['name']};
```

Comment:
What is 'e'? Try to use more descriptive names, here you are mapping a genre

Reply:
I'll use better names for the variables, const and methods, from now on.
Solved